synapseclient/models/team.py

```diff
@@ -1 +1,304 @@
+
+    is_admin: Optional[bool] = None
+    """Whether the given member is an administrator of the team"""
+
+    def fill_from_dict(self, synapse_team_member: Synapse_TeamMember) -> "TeamMember":
+        self.team_id = synapse_team_member.get("teamId", None)
+        self.member = UserGroupHeader().fill_from_dict(
+            synapse_team_member.get("member", None)
+        )
+        self.is_admin = synapse_team_member.get("isAdmin", None)
+        return self
+
+
+@dataclass
+class Team:
+    """
+    Represents a [Synapse Team](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/Team.html).
+    User definable fields are:
+
+    Attributes:
+        id: The ID of the team
+        name: The name of the team
+        description: A short description of the team
+        icon: A file handle ID for the icon image of the team
+        can_public_join: True if members can join without an invitation or approval
+        can_request_membership: True if users can create a membership request to join
+        etag: Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle concurrent updates
+                Since the E-Tag changes every time an entity is updated it is used to detect when
+                a client's current representation of an entity is out-of-date.
+        created_on: The date this team was created
+        modified_on: The date this team was last modified
+        created_by: The ID of the user that created this team
+        modified_by: The ID of the user that last modified this team
+    """
+
+    id: Optional[int] = None
+    """The ID of the team"""
+
+    name: Optional[str] = None
+    """The name of the team"""
+
+    description: Optional[str] = None
+    """A short description of the team"""
+
+    icon: Optional[str] = None
+    """A file handle ID for the icon image of the team"""
+
+    can_public_join: Optional[bool] = None
+    """True if members can join without an invitation or approval"""
+
+    can_request_membership: Optional[bool] = None
+    """True if users can create a membership request to join"""
+
+    etag: Optional[str] = None
+    """Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle concurrent updates
+        Since the E-Tag changes every time an entity is updated it is used to detect when
+        a client's current representation of an entity is out-of-date."""
+
+    created_on: Optional[str] = None
+    """The date this team was created"""
+
+    modified_on: Optional[str] = None
+    """The date this team was last modified"""
+
+    created_by: Optional[str] = None
+    """The ID of the user that created this team"""
+
+    modified_by: Optional[str] = None
+    """The ID of the user that last modified this team"""
+
+    def fill_from_dict(self, synapse_team: Synapse_Team) -> "Team":
+        self.id = (
+            int(synapse_team.get("id", None)) if synapse_team.get("id", None) else None
```

Comment:
It doesn't because it evaluates the `if` part first. It won't try to do the `int` part unless `synapse_team.get("id", None)` is not `None`:
```
team = Team()
team.fill_from_dict(synapse_team={})

Team(id=None, name=None, description=None, icon=None, can_public_join=False, can_request_membership=True, etag=None, created_on=None, modified_on=None, created_by=None, modified_by=None)
```

Reply:
Yeah, you're right. My mistake!